src/braket/circuits/circuit.py

```diff
@@ -877,6 +882,77 @@ def to_ir(self) -> Program:
             basis_rotation_instructions=ir_basis_rotation_instructions,
         )
 
+    @staticmethod
+    def _einsum_subscripts(targets: QubitSet, qubit_count: int) -> str:
+        target_count = len(targets)
+
+        gate_left_indexes = list(range(target_count))
+        un_left_indexes = list(range(target_count, target_count + qubit_count))
+        un_right_indexes = list(range(target_count + qubit_count, target_count + 2 * qubit_count))
+
+        gate_right_indexes = [un_left_indexes[-1 - target] for target in targets]
+
+        result_left_indexes = un_left_indexes.copy()
+        for pos, target in enumerate(targets):
+            result_left_indexes[-1 - target] = gate_left_indexes[pos]
+
+        return (
+            gate_left_indexes + gate_right_indexes,
+            un_left_indexes + un_right_indexes,
+            result_left_indexes + un_right_indexes,
+        )
+
+    def as_unitary(self) -> np.ndarray:
+        """
+        Unitary matrix representation of the entire circuit.
+        Notice that any circuit with 10 qubits or more can get slow.
+
+        Returns:
+            np.ndarray: 2^qubit_count by 2^qubit_count circuit unitary matrix
+
+        Raises:
+            TypeError: If circuit is not composed only of `Gate` instances,
+                i.e. a circuit with `Noise` operators will raise this error.
+            ValueError: If circuit does not have at least 1 qubit.
+        """
+        qubits = self.qubits
+        if not qubits:
+            raise ValueError("Circuit must have at last 1 qubit")
+
+        qubit_count = max(qubits) + 1
+        if qubit_count >= 10:
+            self._logger.warning(
+                "To produce an unitary for 10 or more qubits can get slow. "
+                f"Current count: {qubit_count}"
```

Comment:
```suggestion
                "Unitary calculation for circuits with more than 10 qubits may take longer."
                f"(circuit.qubit_count: {qubit_count})"
```

Reply:
Done in https://github.com/aws/amazon-braket-sdk-python/pull/253/commits/d6a6e15b43fcf52bcf60511d76834446fff93259